### ISSUE TITLE
The forward output of my model created with PyG randomly appears NaN

### ISSUE THREAD

**Issue by YamingZ** [rusty1s/pytorch_geometric#748]

**Comment by YamingZ** [comment]
I have 30 epoch ,NaN appears randomly in one of them，or does not appear until the end

**Comment by tchaton** [comment]
Could you please give more details ? (The code would be nice)

**Comment by YamingZ** [comment]
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = ChebConv(3, 32, K=3)
        self.conv2 = ChebConv(32, 64, K=3)
        self.conv3 = ChebConv(64, 64, K=3)
        self.dense1 = nn.Linear(4096,512)
        self.dense2 = nn.Linear(512,40)

    def forward(self, data):
        pos, batch = data.pos, data.batch
        # generate graph
        radius = 0.2
        edge_index = radius_graph(pos,radius,batch=batch,max_num_neighbors=40)
        edge_weight = utils.RBFWeight(pos,edge_index)
        # feature
        x = utils.SphericalCoordinate(pos)
        # gcn_1
        x = F.relu(self.conv1(x, edge_index, edge_weight=edge_weight, batch=batch, lambda_max=None))
        # down_sampling:512--->128
        idx = fps(pos,batch,ratio=0.5)
        x, pos, batch = x[idx], pos[idx], batch[idx]

        # generate graph
        radius = 0.4
        edge_index = radius_graph(pos, radius, batch=batch, max_num_neighbors=40)
        edge_weight = utils.RBFWeight(pos,edge_index)
        # gcn_2
        x = F.relu(self.conv2(x, edge_index, edge_weight=edge_weight, batch=batch, lambda_max=None))
        # down_sampling:128----->32
        idx = fps(pos,batch,ratio=0.5)
        x, pos, batch = x[idx], pos[idx], batch[idx]

        # generate graph
        radius = 1
        edge_index = radius_graph(pos, radius, batch=batch, max_num_neighbors=40)
        edge_weight = utils.RBFWeight(pos,edge_index)
        # gcn_3
        x = F.relu(self.conv3(x, edge_index, edge_weight=edge_weight, batch=batch, lambda_max=None))
        # down_sampling:32------>16
        idx = fps(pos,batch,ratio=0.5)
        x, pos, batch = x[idx], pos[idx], batch[idx]

        # flatten
        x = x.view(-1, 4096)

        x = F.relu(self.dense1(x))
        x = self.dense2(x)
        # return F.log_softmax(x, dim=1)
        return x

**Comment by YamingZ** [comment]
Some comments are wrong, please don't care

**Comment by YamingZ** [comment]
Some notes are wrong, please don't care

**Comment by rusty1s** [comment]
You can evaluate the cause of NaNs via `torch.autograd.detect_anomaly()`.

**Comment by YamingZ** [comment]
OK,thank you
